src/tools/utils.rs

```diff
@@ -115,3 +119,58 @@ where
     );
     Ok(())
 }
+
+/// block on function
+pub(super) fn block_on<T, F: Future<Output = T>>(f: F) -> Result<T> {
+    Ok(tokio::runtime::Handle::current().block_on(f))
```

Comment:
This implementation will panic if there is no runtime. You should check, whether the runtime is available. If it is then use `tokio::runtime::Handle::current()`, if it is not then create a new runtime